idaes/models/properties/modular_properties/phase_equil/smooth_VLE.py

```diff
@@ -50,116 +50,288 @@ def phase_equil(b, phase_pair):
-                _t1, b.temperature_dew[phase_pair], eps_2
-            )
-
-    elif v_only_comps == []:
+    def rule_temperature_slack_complementarity(b, p):
+        flow_phase = b.flow_mol_phase[p]
+        if b.params.config.supercritical:
+            if p == 'Vap':
+                return smooth_min(s[p], flow_phase, eps) == 0
+            else:
+                return Constraint.Skip
+        else:
+            return smooth_min(s[p], flow_phase, eps) == 0
+    b.add_component(
+        "temperature_slack_complementarity" + suffix,
+        Constraint(
+            b.params.phase_list,
+            rule=rule_temperature_slack_complementarity,
+        ),
+    )
 
-        def rule_teq(b):
-            return b._teq[phase_pair] == _t1
+    def rule_cubic_root_complementarity(b, p):
+        p1, p2 = phase_pair
+        return b.cubic_second_derivative[p1, p2, p] == gp[p] - gn[p]
+    b.add_component(
+        "cubic_root_complementarity" + suffix,
+        Constraint(b.params.phase_list, rule=rule_cubic_root_complementarity),
+    )
 
-    else:
+    def rule_cubic_slack_complementarity(b, p):
+        flow_phase = b.flow_mol_phase[p]
+        if p == 'Vap':
+            return smooth_min(gn[p], flow_phase, eps) == 0
+        else:
+            if b.params.config.supercritical:
+                return smooth_min(gp[p] + s[p], flow_phase, eps) == 0
+            else:
+                return smooth_min(gp[p], flow_phase, eps) == 0
+    b.add_component(
+        "cubic_slack_complementarity" + suffix,
+        Constraint(b.params.phase_list, rule=rule_cubic_slack_complementarity),
+    )
 
-        def rule_teq(b):
-            return b._teq[phase_pair] == b.temperature
+    if b.params.config.supercritical:
+        b.add_component(
+        "pp" + suffix,
+        Var(initialize=0.0,
+            bounds=(0, None),
+            doc="Artificial pressure variable to check if P > Pc",
+            units=pyunits.dimensionless)
+        )
+        pp = getattr(b, "pp" + suffix)
+        
+        b.add_component(
+        "pn" + suffix,
+        Var(initialize=0.0,
+            bounds=(0, None),
+            doc="Artificial pressure variable to check if P < Pc",
+            units=pyunits.dimensionless)
+        )
+        pn = getattr(b, "pn" + suffix)
+    
+    
+        def rule_pbar(b):
+            """
+            pbar = P - Pc * Pm - eps_4 / 4
+            """
+            return b.pbar[phase_pair] == b.pressure - pp - eps / 4
+        b.add_component("pbar_constraint" + suffix, Constraint(rule=rule_pbar))
 
-    b.add_component("_teq_constraint" + suffix, Constraint(rule=rule_teq))
+        # Rule P+
+        def rule_pressure_comparison_1(b):
+            """
+            P+ = max(0, P - Pc)
+            """
+            return pp == smooth_max(0, b.pressure - b.pressure_crit_mix, eps)
+        b.add_component(
+            "eq_pressure_comparison_1" + suffix,
+            Constraint(rule=rule_pressure_comparison_1),
+        )
```

Comment:
Above, `pp` was defined to be dimensionless, but here it appears that it has dimensions of pressure.